apps/admin-gui/src/app/facilities/pages/facility-detail-page/facility-service-status/facility-service-status.component.ts

```diff
@@ -0,0 +1,140 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { ServicesManagerService, ServiceState, TasksManagerService } from '@perun-web-apps/perun/openapi';
+import { SelectionModel } from '@angular/cdk/collections';
+import { PageEvent } from '@angular/material/paginator';
+import {
+  TABLE_FACILITY_SERVICES_STATUS_LIST,
+  TableConfigService
+} from '@perun-web-apps/config/table-config';
+import { NotificatorService } from '@perun-web-apps/perun/services';
+import { TranslateService } from '@ngx-translate/core';
+
+@Component({
+  selector: 'app-facility-service-status',
+  templateUrl: './facility-service-status.component.html',
+  styleUrls: ['./facility-service-status.component.scss']
+})
+export class FacilityServiceStatusComponent implements OnInit {
+
+  constructor(private route: ActivatedRoute,
+              private tasksManager: TasksManagerService,
+              private tableConfigService: TableConfigService,
+              private servicesManager: ServicesManagerService,
+              private notificatior: NotificatorService,
+              private translate: TranslateService) {
+    translate.get('FACILITY_DETAIL.SERVICES_STATUS.SUCCESS_FORCE_PROPAGATION').subscribe(value => this.successFPMessage = value);
+    translate.get('FACILITY_DETAIL.SERVICES_STATUS.SUCCESS_ALLOW').subscribe(value => this.successAllowMessage = value);
+    translate.get('FACILITY_DETAIL.SERVICES_STATUS.SUCCESS_BLOCK').subscribe(value => this.successBlockMessage = value);
+    translate.get('FACILITY_DETAIL.SERVICES_STATUS.SUCCESS_DELETE').subscribe(value => this.successDeleteMessage = value);
+    translate.get('FACILITY_DETAIL.SERVICES_STATUS.ALREADY_DELETED').subscribe(value => this.allreadyDeletedMessage = value);
+  }
+
+  facilityId: number;
+  servicesStates: ServiceState[] = [];
+
+  selected = new SelectionModel<ServiceState>(true, []);
+  filterValue = '';
+  loading: boolean;
+  pageSize: number;
+
+  tableId = TABLE_FACILITY_SERVICES_STATUS_LIST;
+
+  successFPMessage: string;
+  successAllowMessage: string;
+  successBlockMessage: string;
+  successDeleteMessage: string;
+  allreadyDeletedMessage: string
+
+  disableAllowButton = true;
+  disableBlockButton = true;
+  disableDeleteTaskButton = true;
+
+  ngOnInit(): void {
+    this.pageSize = this.tableConfigService.getTablePageSize(this.tableId);
+
+    this.route.parent.params.subscribe(params => {
+      this.facilityId = params['facilityId'];
+      this.refreshTable();
+    });
+  }
+
+  forcePropagation() {
+    for (const ss of this.selected.selected) {
+      this.servicesManager.forceServicePropagation(ss.service.id, ss.facility.id).subscribe(() => {
+        this.refreshTable();
+        this.notificatior.showSuccess(this.successFPMessage);
+        this.selected.clear();
```

Comment:
I guess you dont have to have clear selection in all these methods when you already doing that in refresh table

Reply:
Done